cronex/tests.py

```diff
@@ -422,6 +422,48 @@ def test_unicode_is_accepted_in_python2(self):
             return
         cronex.CronExpression(unicode("* * * * * ABC"))
 
+    def test_vixie_cron_wildcard_flags(self):
+        """
+        Test the arguments do_wild and do_non_wild, which can be used to
+        implement the Vixie cron local time jump tracking algorithm, where
+        non-wildcard and wildcard job execution behavior is changed
+        intelligently in the presence of DST transitions and most forward and
+        reverse clock skew occurrences.
+        """
+
+        # note: normal cron behavior when do_wild=True, do_non_wild=True
+        # other combinations are used during time jump errors
+
+        # matches when do_wild=True
+        testex1 = cronex.CronExpression("* * * * *")
+        # matches when do_non_wild=True
+        testex2 = cronex.CronExpression("0 1 * * *")
+
+        # matches testex1 when do_wild=True and do_non_wild=True
+        tuple1 = (2017, 9, 11, 1, 0)
+        # matches testex1 when do_wild=True only and not testex2
+        tuple2 = (2017, 9, 11, 2, 1)
+
+        self.assertTrue(testex1.check_trigger(tuple1, do_wild=True, do_non_wild=True))
```

Comment:
These tests are simple and repetitive enough that they should be parametrized.

    test_cases = [
        (True, tuple1, True, False),
        # ... repeat for each test ...
    ]

    for expected_result, date, wild, non_wild in test_cases:
        # ...

Reply:
I could do that if it's preferred, it will however cause you to get a stack trace to an arbitrary assertion failure rather than a specific one for a specific failing case.

Reply:
Test assertions allow the user to specify custom messages:

    >>> unittest.TestCase().assertTrue(False, "my message")
    Traceback (most recent call last):
      File "<stdin>", line 1, in <module>
      File "/usr/lib/python3.5/unittest/case.py", line 678, in assertTrue
        raise self.failureException(msg)
    AssertionError: False is not true : my message

You can add the variables to the message using string formatting. I would be fine with you doing `... % locals()` then you can reference the variables inside the format string like `"expected_result = %(expected_result)r, date = %(date)r, ..."`.